python-port/sendpy/usage.py

```diff
@@ -0,0 +1,106 @@
+
+    Send e-mail digitally signed with PGP/MIME
+    $ {programname} --subject "Example" --passphrase "config" --to "User <user@example.com>"
+""")
+
+
+def carriers():
+    """Print out common carriers a user could use to send a text message."""
+    print("SMS and MMS Gateway domains\nSource: https://en.wikipedia.org/wiki/SMS_gateway#Email_clients")
+    print("Use the relevant SMS gateway to send text messages or the MMS gateway to send messages with attachments")
+
+    print("\nUnited States Carriers\n")
+    format_attachment_output([
+        ("Mobile carrier", "SMS gateway domain", "MMS gateway domain"),
+        ("", "", ""),
+        ("Alltel", "sms.alltelwireless.com", "mms.alltelwireless.com"),
+        ("AT&T", "txt.att.net", "mms.att.net"),
+        ("Boost Mobile", "sms.myboostmobile.com", "myboostmobile.com"),
+        ("Cricket Wireless", "mms.cricketwireless.net", "mms.cricketwireless.net"),
+        ("FirstNet", "txt.att.net", "mms.att.net"),
+        ("Google Fi", "msg.fi.google.com", ""),
+        ("MetroPCS", "mymetropcs.com", "mymetropcs.com"),
+        ("Republic Wireless", "text.republicwireless.com", ""),
+        ("Sprint", "messaging.sprintpcs.com", "pm.sprint.com"),
+        ("T-Mobile", "tmomail.net", "tmomail.net"),
+        ("U.S. Cellular", "email.uscc.net", "mms.uscc.net"),
+        ("Verizon Wireless", "vtext.com", "vzwpix.com"),
+        ("Virgin Mobile", "vmobl.com", "vmpix.com")])
+    print("\nCanadian Carriers\n")
+    format_attachment_output([
+        ("Mobile carrier", "SMS gateway domain"),
+        ("", ""),
+        ("Bell Canada", "txt.bell.ca"),
+        ("Bell MTS", "text.mts.net"),
+        ("Fido Solutions", "fido.ca"),
+        ("Freedom Mobile", "txt.freedommobile.ca"),
+        ("Koodo Mobile", "msg.telus.com"),
+        ("PC Mobile", "mobiletxt.ca"),
+        ("Rogers Comm", "pcs.rogers.com"),
+        ("SaskTel", "sms.sasktel.com"),
+        ("Telus", "msg.telus.com")])
+
+
+def servers():
+    """Print out common emails and how to use the SMTP servers to send messages from.
+    This information may be outdated. Information is gained from:
+    https://www.arclab.com/en/kb/email/list-of-smtp-and-pop3-servers-mailserver-list.html.
+    """
+    print("SMTP Servers\nPort 465 typically means SSL/TLS, while Port 587 means StartTLS\n")
+    format_attachment_output([
+        ("Email Provider", "SMTP Server"),
+        ("", ""),
+        ("Gmail", "smtp.gmail.com:465"),
+        ("Outlook", "smtp.live.com:465"),
+        ("Office365", "smtp.office365.com:465"),
+        ("Yahoo Mail", "smtp.mail.yahoo.com:465"),
+        ("Yahoo UK", "smtp.mail.yahoo.co.uk:465"),
+        ("AOL", "smtp.aol.com:587"),
+        ("AT&T", "smtp.att.yahoo.com:587"),
+        ("Hotmail", "smtp.live.com:465 "),
+        ("Comcast", "smtp.comcast.net:587"),
+        ("Verizon", "outgoing.verizon.net:587"),
+        ("Mail.com", "smtp.mail.com:465")])
```

Comment:
```
        ("Gmail", "smtp.gmail.com"),
        ("Outlook", "smtp.live.com"),
        ("Office365", "smtp.office365.com"),
        ("Yahoo Mail", "smtp.mail.yahoo.com"),
        ("Yahoo UK", "smtp.mail.yahoo.co.uk"),
        ("AOL", "smtp.aol.com"),
        ("AT&T", "smtp.att.yahoo.com"),
        ("Hotmail", "smtp.live.com"),
        ("Comcast", "smtp.comcast.net"),
        ("Verizon", "outgoing.verizon.net"),
        ("Mail.com", "smtp.mail.com")])
```

Reply:
The port is still required for StartTLS, as it defaults to port 25, which as you know is probably not what most people would want. I was thinking you might want to add a new column to show which providers use TLS vs StartTLS, although I suspect they all support full SSL/TLS now...

Reply:
I would rather not add something that could be changed/out of date easily, which I suppose that could be if, say, someone didn't support StartTLS and then suddenly did. But if you want to add the column and do that research if you don't think it's a big deal or you want to maintain that, feel free.

Reply:
Sure, I just meant that your `--smtp-servers` option does not provide enough information for users to setup the script, as they would additionally need to know if had to provide either the `--tls` or `--starttls` option. I suspect this list is already outdated, as I am sure all of these common providers have long supported full SSL/TLS and many may even require it...

Reply:
@tdulcet : ![image](https://github.com/tdulcet/Send-Msg-CLI/assets/32185616/f6f36f98-a48b-4e33-8849-6efdbbd8b694)

Reply:
Yes, but users still would need to know if they needed to provide the  `--tls` or `--starttls` option for each provider...

Anyway, I am OK keeping the list as is, but I just thought you should be aware that it does not provide enough information. There is also a similar issue with your `--gateways` option.

Reply:
OK, good to know. I still stick by the principle of leaving some information for them to lookup for their individual use-case, rather than potentially creating confusion by giving them outdated information. Of course, everyone may differ in their approach, but that's my default at least.